From 138d9634ba7c2cdbec810745f250ad40616cb977 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Sat, 6 Jun 2020 12:59:28 +0300
Subject: [PATCH] Don't check $Config{shrplib} in can_compile_static_library()

Debian builds both shared and static variants of libperl so
static linking is supported.

Bug-Debian: https://bugs.debian.org/962319
Forwarded: not-needed
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2020-12-26

---
 lib/ExtUtils/HasCompiler.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/lib/ExtUtils/HasCompiler.pm
+++ b/lib/ExtUtils/HasCompiler.pm
@@ -164,7 +164,9 @@
 	my $output = $args{output} || \*STDOUT;
 
 	my $config = $args{config} || 'ExtUtils::HasCompiler::Config';
-	return if $config->get('useshrplib') eq 'true';
+	# Debian builds both shared and static variants of libperl
+	# see https://bugs.debian.org/962319
+	# return if $config->get('useshrplib') eq 'true';
 
 	my ($source_handle, $source_name) = tempfile('TESTXXXX', DIR => $tempdir, SUFFIX => '.c', UNLINK => 1);
 	my $basename = basename($source_name, '.c');
